### ISSUE TITLE
use simpler feedbase type

### ISSUE THREAD

**Issue by nmushegian** [makerdao/simplecoin#34]
for the time being oracles are going to have closely aligned incentives with owners and will rely on trust anyway - oracles are likely to be picked by hand and can be paid out of band if necessary

Instead there is a role that pushes (value, error, expiration). Could start with actors, then build repeators that read off of 3rd party oracles, then finally change it to work with a generic feed solution when there are actually competing solutions

**Comment by nmushegian** [comment]
see:  https://github.com/nexusdev/feedbase/pull/28

accept the PR and restore simplecoin with PaidFeedbase, then replace it with a normal Feedbase and remove payment functionality.

**Comment by dbrock** [comment]
So we're removing the dependency on feedbase entirely and instead just having the oracles push directly into the simplecoin contract, right?

**Comment by nmushegian** [comment]
yes ideally with same `set(bytes12,bytes32,uint40)` target as feedbase/whatever its called now

**Comment by gbalabasquer** [comment]
@nmushegian Actually what we would need in simplecoin is an interface for `get(bytes12) returns (bytes32, bool) `

To proceed with this change we should deploy feedbase before due the change from uint24 to bytes12.

We could do the deploy, but it would be nice if @dbrock can merge the pull request first.

**Comment by nmushegian** [comment]
@gbalabasquer under this recommendation it would not have an external feedbase dependency at all, it would maintain `key=>value,expiration` and have `set` protected by "oracle" group

**Comment by nmushegian** [comment]
I'm ok with using a `string metadata` just to get this working, but we should transition url -> ipfs gateway url -> ipfs hash as soon as we can

**Comment by dbrock** [comment]
Yeah let's forget about Feedbase for now, just implement this directly in Simplecoin itself.

**Comment by dbrock** [comment]
And disregard @nmushegian's lofty rhetoric about IPFS, we're going with `string`

**Comment by gbalabasquer** [comment]
@nmushegian 
So I understand that there will not be any dependency to feedbase in simplecoin.
We will just do this interface:
`get(bytes12) returns (bytes32, bool)`
The thing is, at the end we will point it to a feedbase address, and the actual contract deployed expects an `uint24` in the `get` function as a parameter, not a `bytes12`.
Maybe it will work anyway, but i'm not sure.

**Comment by nmushegian** [comment]
@dbrock and I need to decide... motivation for `bytes12` is that `address,bytes12` fits in one word (FB source/sink descriptor), motivation for `uint24` is you can split a word like `address,bytes4,uint24`, or `address/sig/id`

**Comment by gbalabasquer** [comment]
@nmushegian 
I've seen you already implemented the new try_get in simplecoin but still using the Feedbase dependency.
Should we remove it now and just add an interface in the simplecoin contract or we will keep the dependency?

**Comment by nmushegian** [comment]
@gbalabasquer Let's keep the dependency and make sure to update it to the latest `Feedbase200` https://github.com/nexusdev/feedbase/blob/master/src/feedbase.sol#L44 

This is what will be used for maker v900
